### PR TITLE
gh-108303: Move tokenize-related data to `Lib/test/tokenizedata`

### PR DESCRIPTION
**Pull request by sobolevn**
In the future, I think, we can move more files to this folder, since there are quite a lot of artefacts for `tokenize` and similar tools.

* Issue: gh-108303

### CODE REVIEW

**Comment by vstinner** [comment]
> Should not badsyntax_3131.py and coding20731.py be included?

I think so, yes.

File: Lib/test/test_tokenize.py [hunk 1]
```diff
@@ -1849,8 +1849,8 @@ def test_random_files(self):
         # pass the '-ucpu' option to process the full directory.
 
         import glob, random
-        fn = support.findfile("tokenize_tests.txt")
-        tempdir = os.path.dirname(fn) or os.curdir
+        fn = support.findfile("tokenize_tests.txt", subdir='tokenizedata')
+        tempdir = os.path.dirname(os.path.dirname(fn)) or os.curdir
```
Comment (vstinner) [inline Lib/test/test_tokenize.py]: I'm not sure about the logical to get tempdir here. Maybe start from __file__ instead? Why is it called tempdir? That's Lib/test/, the directory of test files, no?

Comment (sobolevn) [reply Lib/test/test_tokenize.py]: This is not something I am modifing here. I only changed 
`tempdir = os.path.dirname(fn) or os.curdir` 
to be
`tempdir = os.path.dirname(os.path.dirname(fn)) or os.curdir`

I can open a new issue about it and dig into it later :)

Comment (vstinner) [inline Lib/test/test_tokenize.py]: Well, your PR *does* modify the code. So I suggest simpler code instead:

```suggestion
        tempdir = os.path.dirname(__file__) or os.curdir
```

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by vstinner** [comment]
@serhiy-storchaka: Do you want to review the updated PR? It LGTM.

**Comment by sobolevn** [comment]
@vstinner thanks for your help! 👍

File: Lib/test/test_tokenize.py [hunk 2]
```diff
@@ -1849,8 +1849,8 @@ def test_random_files(self):
         # pass the '-ucpu' option to process the full directory.
 
         import glob, random
-        fn = support.findfile("tokenize_tests.txt")
-        tempdir = os.path.dirname(fn) or os.curdir
+        fn = support.findfile("tokenize_tests.txt", subdir='tokenizedata')
```
Comment (serhiy-storchaka) [inline Lib/test/test_tokenize.py]: Is it even used?

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Should it be backported?

**Comment by sobolevn** [comment]
Similar PRs were not backported:
- https://github.com/python/cpython/pull/108354
- https://github.com/python/cpython/pull/108978

**Comment by vstinner** [comment]
> Should it be backported?

I don't think so. I prefer to minimize backports, and here it's unlikely that we modify test files in stable versions.

**Comment by vstinner** [comment]
> Hi! The buildbot AMD64 Fedora Stable Clang Installed 3.x has failed when building commit 1110c5b.

@sobolevn: Oh, you need to add the new directory to TESTSUBDIRS in Makefile.pre.in. Can you please create a new PR to fix that?

**Comment by vstinner** [comment]
@sobolevn wrote a follow-up fix (merged): PR #109314.

**Pull request merged**